### DISCUSSION TITLE
Configuration questions - no players found

### DISCUSSION THREAD

**Discussion by MattDemers** [music-assistant/support#1511] [Q&A] [answered]
Hey all. I'm having some issues setting this up.

I'm currently running HomeAssistant and Music Assistant Server through Docker on a Synology NAS.

In order to get to the MA web interface, I had to map 8095 to 8096 explicitly in Docker.
I have added successfully MA to HA (I've made sure I'm using the proper beta version) and used NAS' IP with the 8096 port. This added properly.

However, in the MA interface, I have no players (I'm expecting at least two chromecast players). I've also troubleshooted a bit to see if network discovery was the problem; I've made sure that "External Access Enabled" shows in Setting > System > Network in HomeAssistant. I found this referenced [in the FAQ](https://github.com/orgs/music-assistant/discussions/709).

> Make sure the HA internal url is set correctly. Settings --> System --> Network --> Home Assistant URL --> Local network

However, I don't seem to have this menu in MusicAssistant if I'm supposed to. It feels like the right idea to be able to manually specify the HA location, but I'm unsure how to do that.  MusicAssistant's server settings menu have an "X" for "HomeAssistant Addon" so I'm guessing MA isn't talking to HA properly. My only settings are "Providers" "Players" and "Core"; I don't seem to be able to change anything in "Core."

Any thoughts of what I could do?

**Comment by marcelveldt** [comment]
Oops, that documentation is really outdated and more pointed at Music Assistant V2 while you have the V2 beta installed, we should update it.

If you are going to host the Music Assistant Server yourself in Docker you will need to make sure to enable host networking, just some port mappings is not going to work at all.

**Comment by MattDemers** [comment]
Good to know! Unfortunately, I can't seem to get back to the interface while creating a new container in host networking mode.

This is my console output. I've tried creating a new container, new container with a new data directory, etc.

```
2023/09/05 14:00:35 | stdout | OSError: [Errno 105] No buffer space available[0m
2023/09/05 14:00:35 | stdout | listen_socket.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, _value)
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_utils/net.py", line 288, in add_multicast_member
2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/09/05 14:00:35 | stdout | if add_multicast_member(cast(socket.socket, listen_socket), i):
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_utils/net.py", line 373, in create_sockets
2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/09/05 14:00:35 | stdout | listen_socket, respond_sockets = create_sockets(interfaces, unicast, ip_version, apple_p2p=apple_p2p)
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_core.py", line 181, in __init__
2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/09/05 14:00:35 | stdout | self.zeroconf = Zeroconf(interfaces=InterfaceChoice.All)
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/music_assistant/server/server.py", line 105, in start
2023/09/05 14:00:35 | stdout | await mass.start()
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/music_assistant/__main__.py", line 174, in start_mass
2023/09/05 14:00:35 | stdout | await coro
2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/aiorun.py", line 219, in new_coro
2023/09/05 14:00:35 | stdout | Traceback (most recent call last):
2023/09/05 14:00:35 | stdout | future: <Task finished name='Task-1' coro=<run.<locals>.new_coro() done, defined at /usr/local/lib/python3.11/site-packages/aiorun.py:209> exception=OSError(105, 'No buffer space available')>
2023/09/05 14:00:35 | stdout | [31m2023-09-05 19:00:35.260 ERROR (MainThread) [asyncio] Task exception was never retrieved
```

**Reply by OzGav** [reply]
>  don't seem to have this menu in MusicAssistant 

That is in HA settings….

**Reply by OzGav** [reply]
We only have limited ability to answer non-MA questions. We recommend running MA on HAOS to avoid these setup issues. Did you Google that OSERROR?

**Reply by marcelveldt** [reply, marked as answer]
I'm afraid you run into the limitations of using a NAS as docker host. Looks like some networking stack issues now.
Can't you just run Home Assistant OS on some dedicated pi or something or worst case in a VM ?
I've seen the strangest issues with using Synology as docker host, mainly because its underlying OS has all kind of missing features. In the end it is a NAS, not a server OS.

**Reply by MattDemers** [reply]
No worries, that's fine; I'm looking for a replacement for [Spotcast](https://github.com/fondberg/spotcast), and if this doesn't work, that's okay. I might poke around and see if I can run HomeAssistant on a Pi.

**Comment by maste9** [comment]
Was just running into that error after fresh docker install on a ubuntu server 22.04 server based macmini 2012 ...

> ```
> 2023/09/05 14:00:35 | stdout | OSError: [Errno 105] No buffer space available�[0m
> 2023/09/05 14:00:35 | stdout | listen_socket.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, _value)
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_utils/net.py", line 288, in add_multicast_member
> 2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2023/09/05 14:00:35 | stdout | if add_multicast_member(cast(socket.socket, listen_socket), i):
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_utils/net.py", line 373, in create_sockets
> 2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2023/09/05 14:00:35 | stdout | listen_socket, respond_sockets = create_sockets(interfaces, unicast, ip_version, apple_p2p=apple_p2p)
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/zeroconf/_core.py", line 181, in __init__
> 2023/09/05 14:00:35 | stdout | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2023/09/05 14:00:35 | stdout | self.zeroconf = Zeroconf(interfaces=InterfaceChoice.All)
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/music_assistant/server/server.py", line 105, in start
> 2023/09/05 14:00:35 | stdout | await mass.start()
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/music_assistant/__main__.py", line 174, in start_mass
> 2023/09/05 14:00:35 | stdout | await coro
> 2023/09/05 14:00:35 | stdout | File "/usr/local/lib/python3.11/site-packages/aiorun.py", line 219, in new_coro
> 2023/09/05 14:00:35 | stdout | Traceback (most recent call last):
> 2023/09/05 14:00:35 | stdout | future: <Task finished name='Task-1' coro=<run.<locals>.new_coro() done, defined at /usr/local/lib/python3.11/site-packages/aiorun.py:209> exception=OSError(105, 'No buffer space available')>
> 2023/09/05 14:00:35 | stdout | �[31m2023-09-05 19:00:35.260 ERROR (MainThread) [asyncio] Task exception was never retrieved
> ```

... searched for the Error message and found https://github.com/esphome/issues/issues/2669#issuecomment-1436020380

Good News - executing  `sysctl -w net.ipv4.igmp_max_memberships=50` on the host system solved it.

**Comment by ArchonMegalon** [comment]
If someone finds this trying to run this in Docker Desktop on Windows, this is the change to the yaml file you need to add at the end.

**entrypoint: sh -c "sysctl -w net.ipv4.igmp_max_memberships=50 && mass --config /data"**
